From e7f288759f320a958d90b5f003b4d836c4c065cc Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:03:16 -0400
Subject: [usb] wusbcore: set packet count correctly on isoc transfers

Message-id: <1410887463-20674-125-git-send-email-dzickus@redhat.com>
Patchwork-id: 95401
O-Subject: [RHEL7 PATCH 124/591] usb: wusbcore: set packet count correctly on isoc transfers
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 756a2eed67e61e9596c2b49a787441c2e0daf1e1
Author: Thomas Pugliese <thomas.pugliese@gmail.com>
Date:   Mon Dec 9 14:15:15 2013 -0600

    usb: wusbcore: set packet count correctly on isoc transfers

    This patch correctly sets the dwNumOfPackets field of the HWA transfer
    request for isochronous transfers with multiple segments.  Previously
    all segments used the value that was set for the first segment which may
    not be correct.

    Signed-off-by: Thomas Pugliese <thomas.pugliese@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/wusbcore/wa-xfer.c b/drivers/usb/wusbcore/wa-xfer.c
index 6aeb52c..a70e142 100644
--- a/drivers/usb/wusbcore/wa-xfer.c
+++ b/drivers/usb/wusbcore/wa-xfer.c
@@ -1259,8 +1259,11 @@ static int __wa_xfer_setup(struct wa_xfer *xfer, struct urb *urb)
   for (cnt = 1; cnt < xfer->segs; cnt++) {
    struct wa_xfer_packet_info_hwaiso *packet_desc;
    struct wa_seg *seg = xfer->seg[cnt];
+   struct wa_xfer_hwaiso *xfer_iso;
 
    xfer_hdr = &seg->xfer_hdr;
+   xfer_iso = container_of(xfer_hdr,
+      struct wa_xfer_hwaiso, hdr);
    packet_desc = ((void *)xfer_hdr) + xfer_hdr_size;
    /*
     * Copy values from the 0th header. Segment specific
@@ -1270,6 +1273,8 @@ static int __wa_xfer_setup(struct wa_xfer *xfer, struct urb *urb)
    xfer_hdr->bTransferSegment = cnt;
    xfer_hdr->dwTransferLength =
     cpu_to_le32(seg->isoc_size);
+   xfer_iso->dwNumOfPackets =
+     cpu_to_le32(seg->isoc_frame_count);
    __wa_setup_isoc_packet_descr(packet_desc, xfer, seg);
    seg->status = WA_SEG_READY;
   }
-- 
1.7.1